### PR TITLE
Issue 35112/relax cats when not primary key

### PR DESCRIPTION
**Pull request by maxi297**
## What
Addresses https://github.com/airbytehq/airbyte-internal-issues/issues/6377

## How
* Validate that we have at least as many records than the expected
* Remove extra_records
* Remove extra_fields
* Remove basic_read.ignored_fields

### Ignored fields
Removing `ignored_fields` is a bit messy as there are mismatch with `basic_read` tests `full_refresh` which I can't explain. For now, I'll only remove the `basic_read.ignored_fields` but it seems like both should follow the same logic

Script to detect mismatch (to run with cwd in `airbyte-integrations/connectors`):
```
import glob
import yaml


def _get_ignored_fields(test):
    ignored_fields_by_config_path = {}
    for test in test_config.get("acceptance_tests", {}).get(test, {}).get("tests", []):
        if test.get("ignored_fields"):
            ignored_fields_by_config_path[test.get("config_path", "secrets/config.json")] = test.get("ignored_fields")
    return ignored_fields_by_config_path


mismatching_sources = []
for file_name in glob.glob("source-*/acceptance-test-config.yml"):
    with open(file_name) as file:
        test_config = yaml.safe_load(file)
        basic_read_ignored_fields = _get_ignored_fields("basic_read")
        full_refresh_ignored_fields = _get_ignored_fields("full_refresh")
        if not basic_read_ignored_fields == full_refresh_ignored_fields:
            mismatching_sources.append(file_name)

print(mismatching_sources)
print(len(mismatching_sources))
```

Output:
```
['source-linkedin-ads/acceptance-test-config.yml', 'source-shopify/acceptance-test-config.yml', 'source-greenhouse/acceptance-test-config.yml', 'source-tiktok-marketing/acceptance-test-config.yml', 'source-insightly/acceptance-test-config.yml', 'source-facebook-pages/acceptance-test-config.yml', 'source-jira/acceptance-test-config.yml', 'source-coingecko-coins/acceptance-test-config.yml', 'source-klaviyo/acceptance-test-config.yml', 'source-apple-search-ads/acceptance-test-config.yml', 'source-instagram/acceptance-test-config.yml', 'source-square/acceptance-test-config.yml', 'source-gitlab/acceptance-test-config.yml', 'source-amazon-ads/acceptance-test-config.yml', 'source-mixpanel/acceptance-test-config.yml', 'source-coinmarketcap/acceptance-test-config.yml', 'source-amazon-seller-partner/acceptance-test-config.yml', 'source-bing-ads/acceptance-test-config.yml', 'source-freshsales/acceptance-test-config.yml', 'source-google-ads/acceptance-test-config.yml', 'source-sendgrid/acceptance-test-config.yml', 'source-monday/acceptance-test-config.yml', 'source-launchdarkly/acceptance-test-config.yml', 'source-salesforce/acceptance-test-config.yml', 'source-clickup-api/acceptance-test-config.yml', 'source-confluence/acceptance-test-config.yml', 'source-stripe/acceptance-test-config.yml', 'source-slack/acceptance-test-config.yml', 'source-recharge/acceptance-test-config.yml', 'source-google-pagespeed-insights/acceptance-test-config.yml', 'source-paystack/acceptance-test-config.yml', 'source-gcs/acceptance-test-config.yml', 'source-sentry/acceptance-test-config.yml', 'source-airtable/acceptance-test-config.yml', 'source-salesloft/acceptance-test-config.yml', 'source-apify-dataset/acceptance-test-config.yml', 'source-zendesk-talk/acceptance-test-config.yml', 'source-datadog/acceptance-test-config.yml', 'source-pinterest/acceptance-test-config.yml', 'source-hubspot/acceptance-test-config.yml', 'source-github/acceptance-test-config.yml', 'source-google-search-console/acceptance-test-config.yml', 'source-facebook-marketing/acceptance-test-config.yml', 'source-xero/acceptance-test-config.yml']
44
```

If we decide to align the comparison of records in `basic_read` and `full_refresh` or just remove it for full_refresh (see [this Slack discussion](https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1708989100802379)), this would mean that:
* remove `exclude_fields` from `make_hashable`
* remove `delete_fields`

## 🚨 User Impact 🚨
This means that parts of the `acceptance-test-config.yml` are now outdated. There will be a follow up PR to clean them but we want to still deliver value with this. 

Adding a lot of expected records for source-s3 (up to ID=20), we can generate this error:
```
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:test_core.py:1168 Expected to have at least as many records than expected for stream test.
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:test_core.py:1169 missing:
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:conftest.py:344 [
 {
  "id": 10,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 12,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 17,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 19,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 3,
  "name": "asdf",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 1,
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 9,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 15,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 18,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 16,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 20,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 13,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 2,
  "toto": "godo",
  "name": "j4DyXTS7",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 14,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 11,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 }
]
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:test_core.py:1171 expected:
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:conftest.py:344 [
 {
  "id": 10,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 6,
  "name": "Le35Wyic",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 5,
  "name": "77h4aiMP",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 4,
  "name": "1q6jD8Np",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 12,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 17,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 7,
  "name": "xZhh1Kyl",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 19,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 8,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 3,
  "name": "asdf",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 1,
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 9,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 15,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 18,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 16,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 20,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 13,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 2,
  "toto": "godo",
  "name": "j4DyXTS7",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 14,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 11,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 }
]
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:test_core.py:1173 actual:
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:conftest.py:344 [
 {
  "id": 6,
  "name": "Le35Wyic",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 5,
  "name": "77h4aiMP",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 1,
  "name": "PVdhmjb1",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 4,
  "name": "1q6jD8Np",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 3,
  "name": "v0w8fTME",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 2,
  "name": "j4DyXTS7",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 7,
  "name": "xZhh1Kyl",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 8,
  "name": "M2t286iJ",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 }
]
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:test_core.py:1175 extra:
INFO     detailed_logger acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt:conftest.py:344 [
 {
  "id": 1,
  "name": "PVdhmjb1",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 3,
  "name": "v0w8fTME",
  "valid": false,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 },
 {
  "id": 2,
  "name": "j4DyXTS7",
  "valid": true,
  "_ab_source_file_last_modified": "2021-07-25T15:33:04.000000Z",
  "_ab_source_file_url": "simple_test.csv"
 }
]
```

### CODE REVIEW

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/config.py [hunk 1]
```diff
@@ -132,11 +115,6 @@ class IgnoredFieldsConfiguration(BaseConfig):
     bypass_reason: Optional[str] = Field(default=None, description="Reason why this field is considered ignored.")
 
 
-ignored_fields: Optional[Mapping[str, List[IgnoredFieldsConfiguration]]] = Field(
```
Comment (maxi297) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/config.py:135]: As it is not shared anymore for basic_read, the definition has been moved to https://github.com/airbytehq/airbyte/pull/35645/files#diff-6d30b0c7923582524ce6f15dccff0e22689713a162893c6dcd268c6af46f2302R194

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/config.py [hunk 2]
```diff
@@ -214,7 +191,9 @@ class FullRefreshConfig(BaseConfig):
     configured_catalog_path: Optional[str] = configured_catalog_path
     timeout_seconds: int = timeout_seconds
     deployment_mode: Optional[str] = deployment_mode
-    ignored_fields: Optional[Mapping[str, List[IgnoredFieldsConfiguration]]] = ignored_fields
+    ignored_fields: Optional[Mapping[str, List[IgnoredFieldsConfiguration]]] = Field(
```
Comment (maxi297) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/config.py:194]: The "compare records" between "actual" and "expected_records" and between "first read" and "second read" is different hence why we are keeping this. It is worth wondering if those should be aligned but for now, we will assume this is out of scope for this change

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 3]
```diff
@@ -888,13 +887,13 @@ def _validate_records_structure(records: List[AirbyteRecordMessage], configured_
             ), f" Record {record} from {record.stream} stream with fields {record_fields} should have some fields mentioned by json schema: {schema_pathes}"
 
     @staticmethod
-    def _validate_schema(records: List[AirbyteRecordMessage], configured_catalog: ConfiguredAirbyteCatalog, fail_on_extra_columns: Boolean):
```
Comment (maxi297) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:891]: The new field was removed as part of #35556 but was not cleaned up properly

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py [hunk 4]
```diff
@@ -54,6 +54,10 @@ def assert_emitted_at_increase_on_subsequent_runs(self, first_read_records, seco
     def assert_two_sequential_reads_produce_same_or_subset_records(
         self, records_1, records_2, configured_catalog, ignored_fields, detailed_logger
     ):
+        # should we re-use how we compare records at this point or just completely remove this assertion? Implications:
```
Comment (brianjlai) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py:57]: I'm fine with killing this

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 5]
```diff
@@ -1179,76 +1155,27 @@ def compare_records(
                     not expected_but_not_found
                 ), f"Expected to see those primary keys in the actual response for stream {stream_name} but they were not found."
         else:
-            TestBasicRead.legacy_compare_records(
-                stream_name, actual, expected, extra_fields, exact_order, extra_records, ignored_fields, detailed_logger
-            )
+            if len(expected) > len(actual):
```
Comment (brianjlai) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:1158]: nit: It looks like the rest of the logging and fail code is all encompassed in this `if` conditional. Can we just combine the above `else` into `elif len(expected > len(actual)`

**Review by brianjlai — APPROVED** [review]
lgtm, just small things and responses to your comments

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py (hunk 5 above)
Comment (maxi297) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:1158]: You are right. In my mind, it was "if primary_key: do X, else: do Y" but the flow makes more sense as you describe

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py (hunk 4 above)
Comment (maxi297) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py:57]: It is killed. I'll organize the funerals
